### PR TITLE
fix(corel): update uses of collate, add typeguard for bundles and add release tag to addonDatasetProvider

### PR DESCRIPTION
**Pull request by pedrobonamin**
Fixes build issues in COREL branch.

### CODE REVIEW

File: packages/sanity/src/core/form/studio/inputs/client-adapters/reference.ts [hunk 1]
```diff
@@ -201,7 +201,7 @@ export function referenceSearch(
   })
   return search(textTerm, {includeDrafts: true}).pipe(
     map(({hits}) => hits.map(({hit}) => hit)),
-    map(collate),
+    map((docs) => collate(docs)),
```
Comment (jordanl17) [inline packages/sanity/src/core/form/studio/inputs/client-adapters/reference.ts:204]: Q: so this wasn't previously working...?

**Comment by jordanl17** [comment]
@pedrobonamin seems to still be some linting errors. Is the intention of this change to resolve that also?

File: packages/sanity/src/core/form/studio/inputs/client-adapters/reference.ts (hunk 1 above)
Comment (pedrobonamin) [reply packages/sanity/src/core/form/studio/inputs/client-adapters/reference.ts:204]: It was working, types were throwing.
in a map, the first parameter is the item, the second is a number with the index.

When doing  `map(collate)` it's the same as doing `map((item, index) => collate(item, index))`

And collate was updated to receive an object as second parameter https://github.com/sanity-io/sanity/blob/fix-build/packages/sanity/src/core/util/draftUtils.ts#L142-L143

That's why it was showing a ts error

Comment (jordanl17) [reply packages/sanity/src/core/form/studio/inputs/client-adapters/reference.ts:204]: Makes sense! Thanks

**Review by jordanl17 — APPROVED** [review]
Nice 🔥

**Review by RitaDias — APPROVED** [review]
🚀 let's go

**Pull request merged**